integrations/instagram/src/misc/utils.ts

```diff
@@ -62,18 +71,43 @@ export function getCarouselMessage(payload: Carousel): MessengerTypes.Attachment
   }
 }
 
-export function getChoiceMessage(payload: Choice | Dropdown): MessengerTypes.AttachmentMessage {
+export function getChoiceMessage(payload: Choice | Dropdown): MessengerTypes.TextMessage {
+  if (!payload.options.length) {
+    return { text: payload.text }
+  }
+
+  if (payload.options.length > 13) {
+    return {
+      text: `${payload.text}\n\n${payload.options.map((o, idx) => `${idx + 1}. ${o.label}`).join('\n')}`,
+    }
+  }
+
   return {
-    attachment: {
-      type: 'template',
-      payload: {
-        templateType: 'generic',
-      },
-    },
-    quickReplies: payload.options.map((choice) => ({
+    text: payload.text,
+    quickReplies: payload.options.map((option) => ({
       contentType: 'text',
-      title: choice.label,
-      payload: choice.value,
+      title: option.label,
+      payload: option.value,
     })),
   }
 }
+
+export const getUserProfile = async (
+  userId: string,
+  configuration: bp.configuration.Configuration,
+  logger: IntegrationLogger
+) => {
+  const messengerClient = getMessengerClient(configuration)
+  try {
+    return (await messengerClient.getUserProfile(userId, {
+      // username is an available field for instagram ids -> https://developers.facebook.com/docs/instagram-basic-display-api/guides/getting-profiles-and-media
+      fields: ['id', 'name', 'profile_pic', 'username'] as any,
+    })) as InstagramUserProfile
+  } catch (error) {
+    logger.forBot().debug("profile_pic can't be fetched from instagram, trying without it")
+    // if the user is not a business instagram user, this will fail because of profile_pic
+    return (await messengerClient.getUserProfile(userId, {
```

Comment:
I think we want to nest this fallback to in its own try/catch to avoid the message processing from breaking if the simple profile cannot be fetched for any reason? (e.g. transient network issue)

Reply:
Ah right, I see that now 👍